src/main/java/ch/hftm/controllers/SceneController.java

```diff
@@ -31,4 +31,20 @@ public <T> void changeScene(Event event, String sceneName, T data) throws IOExce
         stage.show();
     }
 
+    public <T> void changeScene(Scene scene, String sceneName, T data) throws IOException {
+        FXMLLoader fxmlLoader = new FXMLLoader(App.class.getResource(String.format("view/%s.fxml", sceneName)));
+        root = fxmlLoader.load();
+
+        var controller = fxmlLoader.getController();
+        if (controller instanceof AlbumController albumController && data instanceof ch.hftm.data.Album album) {
+            albumController.setAlbum(album);
+        } else if (controller instanceof ImageController imageController && data instanceof ch.hftm.data.Image image) {
+            imageController.setImage(image);
+        }
+
+        stage = (Stage) scene.getWindow();
+        stage.setScene(new Scene(root));
+        stage.show();
+    }
```

Comment:
Actually this makes more sense to pass the scene instead of the event, I refactored the method which opens the album view from the library. Now only one changeScene method is used. I pushed the changes.

Reply:
Adding the preview features results in errors on my setup. Apparently preview features can only be enabled at source level 19. Can't we use an if-else statement instead?